### PR TITLE
fix accounts index startup stat and add two new stats for dups

### PR DESCRIPTION
**Pull request by HaoranYi**
#### Problem

1. In accounts index startup stats, "total_item, max_bin_size, min_bin_size" are not populated. We are reading these stats too early - before the index are actually flushed. That's why they are all zeros. We should wait to read them after flush.

```
[2024-10-08T14:43:43.225032940Z
INFO
solana_metrics::metrics]
datapoint:
generate_index
overall_us=590639563i
total_us=505808369i
scan_stores_us=7810839i
insertion_time_us=67372816i
min_bin_size=0i
max_bin_size=0i
storage_size_storages_us=92035i
index_flush_us=424152i
total_rent_paying=0i
amount_to_top_off_rent=0i
total_items_including_duplicates=650345708i
total_items=0i
accounts_data_len_dedup_time_us=81218852i
total_duplicate_slot_keys=51300596i
populate_duplicate_keys_us=3019104i
total_slots=433370i
slots_to_clean=411148i

```

2.  "total_item, max_bin_size, min_bin_size" actually represent the index items stored inside in-memory cache. Rename them to make it explicit that they are stats for in-memory index.

3. Add a stats to count how many dups (accounts_duplicates_num) and total_num_unique_duplicate_keys encountered during create account index at startup.

#### Summary of Changes

- fix account index startup stats
- add stats for dup account num at startup

```
[2024-10-08T20:12:36.127914984Z
INFO
solana_metrics::metrics]
datapoint:
generate_index
overall_us=145751031i
total_us=87320399i
scan_stores_us=1163008i
insertion_time_us=98630200i
min_bin_size_in_mem=989i
max_bin_size_in_mem=68834i
storage_size_storages_us=78822i
index_flush_us=525962i
total_rent_paying=0i
amount_to_top_off_rent=0i
total_items_including_duplicates=665224689i
total_items_in_mem=10070323i
accounts_data_len_dedup_time_us=55835509i
total_duplicate_slot_keys=52628546i
accounts_duplicates_num=42558223i
populate_duplicate_keys_us=1942864i
total_slots=418500i
slots_to_clean=418406i
```

Fixes #

### CODE REVIEW

File: accounts-db/src/accounts_db.rs [hunk 1]
```diff
@@ -739,6 +741,16 @@ impl GenerateIndexTimings {
...
                 i64
             ),
+            (
+                "total_num_unique_duplicate_keys",
+                self.total_num_unique_duplicate_keys as i64,
+                i64
+            ),
+            (
+                "accounts_duplicates_num",
+                self.accounts_duplicates_num as i64,
```
Comment (brooksprumo) [inline accounts-db/src/accounts_db.rs]: Can this be `num_duplicate_accounts` instead?

File: accounts-db/src/accounts_db.rs [hunk 2]
```diff
@@ -8641,8 +8673,9 @@ impl AccountsDb {
         pubkeys: &[Pubkey],
         rent_collector: &RentCollector,
         timings: &GenerateIndexTimings,
-    ) -> (u64, IntSet<Slot>) {
+    ) -> (u64, u64, IntSet<Slot>) {
```
Comment (brooksprumo) [inline accounts-db/src/accounts_db.rs:8676]: Since the return type is changing, can you please update the function's doc comments to reflect this?

**Review by brooksprumo — COMMENTED** [review]
I think the code looks good. Some nits and requests to rename datapoints, since they are notoriously hard to change later.

File: accounts-db/src/accounts_db.rs [hunk 3]
```diff
@@ -672,16 +672,18 @@ struct GenerateIndexTimings {
...
+    pub total_items_in_mem: usize,
     pub storage_size_storages_us: u64,
     pub index_flush_us: u64,
     pub rent_paying: AtomicUsize,
     pub amount_to_top_off_rent: AtomicU64,
     pub total_including_duplicates: u64,
     pub accounts_data_len_dedup_time_us: u64,
     pub total_duplicate_slot_keys: u64,
+    pub total_num_unique_duplicate_keys: u64,
+    pub accounts_duplicates_num: u64,
```
Comment (brooksprumo) [inline accounts-db/src/accounts_db.rs]: I kinda wish this was in a separate PR. So one PR for fixing the existing stats, and a second PR for adding the new stats.

**Review by brooksprumo — APPROVED** [review]
Looks good. I think this is correct.

**Review by jeffwashington — APPROVED** [review]
lgtm

**Pull request merged**